File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.py
class Solution:
    def searchRange(self, nums: List[int], target: int) -> List[int]:
        first = bisect_left(nums,target)
        second = bisect_right(nums,target) -1
        
        # print(first,second)
        if first >= len(nums) or second >= len(nums):
            return [-1, -1]
        elif nums[first] != target or nums[second] != target:
            return [-1,-1]
        
        return [first,second]
        

File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.cpp
class Solution {
public:
    vector<int> searchRange(vector<int>& nums, int target) {
        int last = upper_bound(nums.begin(), nums.end(), target) - nums.begin();
        int start = lower_bound(nums.begin(), nums.end(), target) - nums.begin();
        
        if(start < nums.size() && nums[start] == target){
            return {start, last-1};
        }
        
    
        return {-1,-1};
    }
};